components/brave_shields/core/browser/brave_shields_utils.cc

```diff
@@ -921,4 +921,60 @@ bool IsDeveloperModeEnabled(PrefService* profile_state) {
   return profile_state->GetBoolean(prefs::kAdBlockDeveloperMode);
 }
 
+// Builds a `base::Value::Dict` from the given `AutoShredType`.
+// This should be used in conjuction with `GetAutoShredTypeFromDict`.
+base::Value GetDictFromAutoShredType(AutoShredType type) {
+  base::Value dict(base::Value::Type::DICT);
+  dict.GetDict().Set("value", static_cast<int>(type));
+  return dict;
+}
+
+/// Finds the `AutoShredType` from the given single-value dict.
+/// This should be used in conjunction with `GetDictFromAutoShredType`.
+AutoShredType GetAutoShredTypeFromDict(base::Value dict) {
+  if (!dict.is_dict()) {
+    return AutoShredType::NEVER;
+  }
+
+  const base::Value* value = dict.GetDict().Find("value");
```

Comment:
Could it be `braveAutoShred` instead of `value`? then I'll be able to use the approach from my PR (cosmetic filters) without adding any migration code.

Reply:
It's just a wish, because I always can leave `value` in specialization

Reply:
I think it's best to update this PR to use the new `BraveShieldsSettings` in https://github.com/brave/brave-core/pull/30839 once it lands. It currently follows the same approach of storing an enum (`ControlType`) in a dictionary as a website setting, but has a more reusable approach with the templates that can be re-used here with the `AutoShredType` enum instead of `ControlType`.